doc/cmake_application.md

```diff
@@ -0,0 +1,42 @@
+# SDK CMake integration with your application
```

Comment:
I would only suggest renaming the file to something like "how_to_use_azure_iot_sdk_c_with_cmake.md", to be more intuitive.